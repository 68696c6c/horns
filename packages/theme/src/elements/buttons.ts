import { BorderStyle, evalSideBordersConfig } from '../borders'
import { Colorway } from '../colors'
import { Cursor } from '../cursors'
import { evalCornerSizesConfig, evalSideSizesConfig, Size } from '../sizes'
import { Font } from '../typography'

import { ElementConfig, ElementTheme } from './elements'

export const defaultButtons: ElementConfig = {
  color: Colorway.Background,
  cursor: Cursor.Pointer,
  border: {
    all: {
      width: Size.Tiny,
      style: BorderStyle.Solid,
    },
  },
  font: Font.Button,
  padding: {
    x: Size.Medium,
    y: Size.Small,
  },
  radius: {
    all: Size.Tiny,
  },
}

export const makeButtons = (config?: Partial<ElementConfig>): ElementTheme => ({
  color: config?.color || defaultButtons.color,
  cursor: config?.cursor || defaultButtons.cursor,
  border: evalSideBordersConfig(defaultButtons.border, config?.border),
  font: config?.font || defaultButtons.font,
  padding: evalSideSizesConfig(defaultButtons.padding, config?.padding),
  radius: evalCornerSizesConfig(defaultButtons.radius, config?.radius),
})
