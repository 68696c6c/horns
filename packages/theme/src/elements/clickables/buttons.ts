import {
  BorderStyle,
  Color,
  evalCornerSizesConfigs,
  evalShadows,
  evalSideBordersConfigs,
  evalSideSizesConfigs,
  Font,
  ShadowType,
  Size,
} from '../../config'
import { Cursor, DeepPartial, evalConfigBooleans } from '../../utils'

import * as Clickable from './clickable'

export const defaultConfig: Required<Clickable.Config> = {
  border: {
    all: {
      width: Size.Tiny,
      style: BorderStyle.Solid,
    },
  },
  color: Color.BgPrimary,
  cursor: Cursor.Pointer,
  font: Font.Control,
  outlined: false,
  padding: {
    x: Size.Medium,
    y: Size.Small,
  },
  radius: {
    all: Size.Tiny,
  },
  shadowed: false,
  shadowType: ShadowType.Box,
  typographic: false,
  variant: 'button',
}

export const make = (
  input?: DeepPartial<Clickable.Config>,
): Clickable.Theme => ({
  border: evalSideBordersConfigs(defaultConfig.border, input?.border),
  color: input?.color || defaultConfig.color,
  cursor: input?.cursor || defaultConfig.cursor,
  font: input?.font || defaultConfig.font,
  outlined: evalConfigBooleans(defaultConfig.outlined, 'outlined', input),
  padding: evalSideSizesConfigs(defaultConfig.padding, input?.padding),
  shadow: evalShadows(defaultConfig.shadowType, defaultConfig.shadowed, input),
  radius: evalCornerSizesConfigs(defaultConfig.radius, input?.radius),
  typographic: defaultConfig.typographic,
  variant: input?.variant || defaultConfig.variant,
})
