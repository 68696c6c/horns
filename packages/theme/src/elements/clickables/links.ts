import {
  BorderStyle,
  Color,
  evalCornerSizesConfigs,
  evalShadows,
  evalSideBordersConfigs,
  evalSideSizesConfigs,
  Font,
  ShadowType,
  Size,
} from '../../config'
import { Cursor, DeepPartial, evalConfigBooleans } from '../../utils'

import * as Clickable from './clickable'

export const defaultConfig: Clickable.Config = {
  border: {
    all: {
      width: Size.None,
      style: BorderStyle.None,
    },
  },
  color: Color.Action,
  cursor: Cursor.Pointer,
  font: Font.Link,
  outlined: false,
  padding: {
    all: Size.None,
  },
  radius: {
    all: Size.None,
  },
  shadowed: false,
  shadowType: ShadowType.Text,
  typographic: true,
  variant: 'link',
}

export const make = (
  input?: DeepPartial<Clickable.Config>,
): Clickable.Theme => ({
  border: evalSideBordersConfigs(defaultConfig.border, input?.border),
  color: input?.color || defaultConfig.color,
  cursor: input?.cursor || defaultConfig.cursor,
  font: input?.font || defaultConfig.font,
  outlined: evalConfigBooleans(defaultConfig.outlined, 'outlined', input),
  padding: evalSideSizesConfigs(defaultConfig.padding, input?.padding),
  radius: evalCornerSizesConfigs(defaultConfig.radius, input?.radius),
  shadow: evalShadows(defaultConfig.shadowType, defaultConfig.shadowed, input),
  typographic: defaultConfig.typographic,
  variant: input?.variant || defaultConfig.variant,
})
