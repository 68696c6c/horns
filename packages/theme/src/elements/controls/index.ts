import {
  BorderStyle,
  Color,
  ControlColor,
  evalCornerSizesConfigs,
  evalShadows,
  evalSideBordersConfigs,
  evalSideSizesConfigs,
  Font,
  ShadowType,
  Size,
} from '../../config'
import { Cursor, DeepPartial, evalConfigBooleans } from '../../utils'

import * as Interactive from '../interactive'

type Control<
  BaseType extends Interactive.Config | Interactive.Props | Interactive.Theme,
> = Omit<BaseType, 'color'> & {
  color: ControlColor
}

export interface Config extends Control<Interactive.Config> {}

export interface Props extends Control<Interactive.Props> {}

export interface Theme extends Control<Interactive.Theme> {}

export const defaultConfig: Config = {
  border: {
    all: {
      width: Size.Tiny,
      style: BorderStyle.Solid,
    },
  },
  color: Color.BgPrimary,
  cursor: Cursor.Default,
  font: Font.Control,
  outlined: false,
  padding: {
    all: Size.XSmall,
  },
  radius: {
    all: Size.Tiny,
  },
  shadowed: false,
  shadowType: ShadowType.Inset,
  typographic: false,
}

export const make = (input?: DeepPartial<Config>): Theme => ({
  border: evalSideBordersConfigs(defaultConfig.border, input?.border),
  color: input?.color || defaultConfig.color,
  cursor: input?.cursor || defaultConfig.cursor,
  font: input?.font || defaultConfig.font,
  outlined: evalConfigBooleans(defaultConfig.outlined, 'outlined', input),
  padding: evalSideSizesConfigs(defaultConfig.padding, input?.padding),
  radius: evalCornerSizesConfigs(defaultConfig.radius, input?.radius),
  shadow: evalShadows(defaultConfig.shadowType, defaultConfig.shadowed, input),
  typographic: defaultConfig.typographic,
})
