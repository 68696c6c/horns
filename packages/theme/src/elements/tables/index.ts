import {
  Breakpoint,
  Color,
  evalCornerSizesConfigs,
  evalShadows,
  evalSideBordersConfigs,
  evalSideSizesConfigs,
  Font,
} from '../../config'
import { DeepPartial, evalConfigBooleans } from '../../utils'

import * as Interactive from '../interactive'

import * as Item from './item'

type Table<ItemType> = ItemType & {
  breakpoint: Breakpoint
  headerItem: DeepPartial<ItemType>
}

export interface Config extends Table<Interactive.Items<Item.Config>> {}

export interface Props extends Table<Interactive.Items<Item.Props>> {}

export interface Theme extends Table<Interactive.Items<Item.Theme>> {}

export const defaultConfig: Config = {
  ...Item.defaultConfig,
  breakpoint: Breakpoint.Mobile,
  currentItem: {
    color: Color.BgSecondary,
  },
  headerItem: {
    font: Font.Strong,
  },
}

const makeItem = (
  base: Item.Theme,
  input?: DeepPartial<Config>,
  item?: DeepPartial<Item.Config>,
): Item.Theme => ({
  border: evalSideBordersConfigs(base.border, input?.border, item?.border),
  color: item?.color || base.color,
  cursor: item?.cursor || base.cursor,
  font: item?.font || base.font,
  outlined: evalConfigBooleans(base.outlined, 'outlined', item),
  padding: evalSideSizesConfigs(base.padding, item?.padding),
  radius: evalCornerSizesConfigs(base.radius, item?.radius),
  shadow: evalShadows(
    defaultConfig.shadowType,
    defaultConfig.shadowed,
    input,
    item,
  ),
  typographic: base.typographic,
})

export const make = (input?: DeepPartial<Config>): Theme => {
  const base = Item.make(input)
  return {
    ...base,
    breakpoint: input?.breakpoint || defaultConfig.breakpoint,
    currentItem: makeItem(base, input, input?.currentItem),
    headerItem: makeItem(base, input, input?.headerItem),
  }
}
