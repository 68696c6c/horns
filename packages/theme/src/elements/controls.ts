import { BorderStyle, evalSideBordersConfigs } from '../borders'
import { Colorway } from '../colors'
import { Cursor } from '../cursors'
import { evalCornerSizesConfigs, evalSideSizesConfigs, Size } from '../sizes'
import { Font } from '../typography'

import { ElementConfig, ElementTheme } from './elements'

export const defaultControls: ElementConfig = {
  color: Colorway.Background,
  cursor: Cursor.Default,
  border: {
    all: {
      width: Size.Tiny,
      style: BorderStyle.Solid,
    },
  },
  font: Font.Control,
  padding: {
    all: Size.XSmall,
  },
  radius: {
    all: Size.Tiny,
  },
}

export const makeControls = (
  config?: Partial<ElementConfig>,
): ElementTheme => ({
  color: config?.color || defaultControls.color,
  cursor: config?.cursor || defaultControls.cursor,
  border: evalSideBordersConfigs(defaultControls.border, config?.border),
  font: config?.font || defaultControls.font,
  padding: evalSideSizesConfigs(defaultControls.padding, config?.padding),
  radius: evalCornerSizesConfigs(defaultControls.radius, config?.radius),
})
