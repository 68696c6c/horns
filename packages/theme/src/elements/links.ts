import { BorderStyle, evalSideBordersConfig } from '../borders'
import { Colorway } from '../colors'
import { Cursor } from '../cursors'
import { evalCornerSizesConfig, evalSideSizesConfig, Size } from '../sizes'
import { Font } from '../typography'

import { ElementConfig, ElementTheme } from './elements'

export const defaultLinks: ElementConfig = {
  color: Colorway.Background,
  cursor: Cursor.Pointer,
  border: {
    all: {
      width: Size.None,
      style: BorderStyle.None,
    },
  },
  font: Font.Link,
  padding: {
    all: Size.None,
  },
  radius: {
    all: Size.None,
  },
}

export const makeLinks = (config?: Partial<ElementConfig>): ElementTheme => ({
  color: config?.color || defaultLinks.color,
  cursor: config?.cursor || defaultLinks.cursor,
  border: evalSideBordersConfig(defaultLinks.border, config?.border),
  font: config?.font || defaultLinks.font,
  padding: evalSideSizesConfig(defaultLinks.padding, config?.padding),
  radius: evalCornerSizesConfig(defaultLinks.radius, config?.radius),
})
