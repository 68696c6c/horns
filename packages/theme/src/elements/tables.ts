import { BorderStyle, evalSideBordersConfig } from '../borders'
import { Colorway } from '../colors'
import { Cursor } from '../cursors'
import { evalCornerSizesConfig, evalSideSizesConfig, Size } from '../sizes'
import { Font } from '../typography'

import { ElementConfig, ElementTheme } from './elements'

export const defaultTables: ElementConfig = {
  color: Colorway.Background,
  cursor: Cursor.Default,
  border: {
    all: {
      width: Size.Tiny,
      style: BorderStyle.Solid,
    },
  },
  font: Font.Button,
  padding: {
    all: Size.XSmall,
  },
  radius: {
    all: Size.Tiny,
  },
}

export const makeTables = (config?: Partial<ElementConfig>): ElementTheme => ({
  color: config?.color || defaultTables.color,
  cursor: config?.cursor || defaultTables.cursor,
  border: evalSideBordersConfig(defaultTables.border, config?.border),
  font: config?.font || defaultTables.font,
  padding: evalSideSizesConfig(defaultTables.padding, config?.padding),
  radius: evalCornerSizesConfig(defaultTables.radius, config?.radius),
})
