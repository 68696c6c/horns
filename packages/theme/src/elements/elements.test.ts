import { BorderStyle } from '../borders'
import { Colorway } from '../colors'
import { Size } from '../sizes'
import { Font } from '../typography'

import { ElementTheme, makeButtons, makeControls, makeTables } from '.'
import { defaultButtons } from './buttons'
import { defaultControls } from './controls'
import { defaultTables } from './tables'
import { defaultLinks, makeLinks } from './links'

describe('makeButtons', () => {
  let expected: ElementTheme
  beforeEach(() => {
    expected = makeButtons(defaultButtons)
  })
  it('should return default values if no input is provided', () => {
    const result = makeButtons()
    expect(result).toStrictEqual(expected)
  })
  it('should accept "all" inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        all: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        all: Size.Giant,
      },
      radius: {
        all: Size.Giant,
      },
    }
    const result = makeButtons(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept dimensional inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        x: { style: BorderStyle.Dotted, width: Size.Giant },
        y: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        x: Size.Giant,
        y: Size.Giant,
      },
      radius: {
        top: Size.Large,
        bottom: Size.XLarge,
        right: Size.XXLarge,
        left: Size.Giant,
      },
    }
    const result = makeButtons(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept complete inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        top: { style: BorderStyle.Dotted, width: Size.Giant },
        bottom: { style: BorderStyle.Dotted, width: Size.Giant },
        left: { style: BorderStyle.Dotted, width: Size.Giant },
        right: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        top: Size.Giant,
        bottom: Size.Giant,
        left: Size.Giant,
        right: Size.Giant,
      },
      radius: {
        topRight: Size.Giant,
        bottomRight: Size.Giant,
        topLeft: Size.Giant,
        bottomLeft: Size.Giant,
      },
    }
    const result = makeButtons(input)
    expect(result).toMatchSnapshot()
  })
})

describe('makeControls', () => {
  let expected: ElementTheme
  beforeEach(() => {
    expected = makeControls(defaultControls)
  })
  it('should return default values if no input is provided', () => {
    const result = makeControls()
    expect(result).toStrictEqual(expected)
  })
  it('should accept "all" inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        all: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        all: Size.Giant,
      },
      radius: {
        all: Size.Giant,
      },
    }
    const result = makeControls(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept dimensional inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        x: { style: BorderStyle.Dotted, width: Size.Giant },
        y: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        x: Size.Giant,
        y: Size.Giant,
      },
      radius: {
        top: Size.Large,
        bottom: Size.XLarge,
        right: Size.XXLarge,
        left: Size.Giant,
      },
    }
    const result = makeControls(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept complete inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        top: { style: BorderStyle.Dotted, width: Size.Giant },
        bottom: { style: BorderStyle.Dotted, width: Size.Giant },
        left: { style: BorderStyle.Dotted, width: Size.Giant },
        right: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        top: Size.Giant,
        bottom: Size.Giant,
        left: Size.Giant,
        right: Size.Giant,
      },
      radius: {
        topRight: Size.Giant,
        bottomRight: Size.Giant,
        topLeft: Size.Giant,
        bottomLeft: Size.Giant,
      },
    }
    const result = makeControls(input)
    expect(result).toMatchSnapshot()
  })
})

describe('makeLinks', () => {
  let expected: ElementTheme
  beforeEach(() => {
    expected = makeLinks(defaultLinks)
  })
  it('should return default values if no input is provided', () => {
    const result = makeLinks()
    expect(result).toStrictEqual(expected)
  })
  it('should accept "all" inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        all: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        all: Size.Giant,
      },
      radius: {
        all: Size.Giant,
      },
    }
    const result = makeLinks(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept dimensional inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        x: { style: BorderStyle.Dotted, width: Size.Giant },
        y: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        x: Size.Giant,
        y: Size.Giant,
      },
      radius: {
        top: Size.Large,
        bottom: Size.XLarge,
        right: Size.XXLarge,
        left: Size.Giant,
      },
    }
    const result = makeLinks(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept complete inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        top: { style: BorderStyle.Dotted, width: Size.Giant },
        bottom: { style: BorderStyle.Dotted, width: Size.Giant },
        left: { style: BorderStyle.Dotted, width: Size.Giant },
        right: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        top: Size.Giant,
        bottom: Size.Giant,
        left: Size.Giant,
        right: Size.Giant,
      },
      radius: {
        topRight: Size.Giant,
        bottomRight: Size.Giant,
        topLeft: Size.Giant,
        bottomLeft: Size.Giant,
      },
    }
    const result = makeLinks(input)
    expect(result).toMatchSnapshot()
  })
})

describe('makeTables', () => {
  let expected: ElementTheme
  beforeEach(() => {
    expected = makeTables(defaultTables)
  })
  it('should return default values if no input is provided', () => {
    const result = makeTables()
    expect(result).toStrictEqual(expected)
  })
  it('should accept "all" inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        all: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        all: Size.Giant,
      },
      radius: {
        all: Size.Giant,
      },
    }
    const result = makeTables(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept dimensional inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        x: { style: BorderStyle.Dotted, width: Size.Giant },
        y: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        x: Size.Giant,
        y: Size.Giant,
      },
      radius: {
        top: Size.Large,
        bottom: Size.XLarge,
        right: Size.XXLarge,
        left: Size.Giant,
      },
    }
    const result = makeTables(input)
    expect(result).toMatchSnapshot()
  })
  it('should accept complete inputs', () => {
    const input = {
      color: Colorway.Primary,
      border: {
        top: { style: BorderStyle.Dotted, width: Size.Giant },
        bottom: { style: BorderStyle.Dotted, width: Size.Giant },
        left: { style: BorderStyle.Dotted, width: Size.Giant },
        right: { style: BorderStyle.Dotted, width: Size.Giant },
      },
      font: Font.Paragraph,
      padding: {
        top: Size.Giant,
        bottom: Size.Giant,
        left: Size.Giant,
        right: Size.Giant,
      },
      radius: {
        topRight: Size.Giant,
        bottomRight: Size.Giant,
        topLeft: Size.Giant,
        bottomLeft: Size.Giant,
      },
    }
    const result = makeTables(input)
    expect(result).toMatchSnapshot()
  })
})
