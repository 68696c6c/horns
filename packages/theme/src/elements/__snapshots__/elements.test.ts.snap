// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeButtons should accept "all" inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "pointer",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeButtons should accept complete inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "pointer",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeButtons should accept dimensional inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "pointer",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "xLarge",
    "bottomRight": "xLarge",
    "topLeft": "large",
    "topRight": "large",
  },
}
`;

exports[`makeControls should accept "all" inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "default",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeControls should accept complete inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "default",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeControls should accept dimensional inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "default",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "xLarge",
    "bottomRight": "xLarge",
    "topLeft": "large",
    "topRight": "large",
  },
}
`;

exports[`makeLinks should accept "all" inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "pointer",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeLinks should accept complete inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "pointer",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeLinks should accept dimensional inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "pointer",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "xLarge",
    "bottomRight": "xLarge",
    "topLeft": "large",
    "topRight": "large",
  },
}
`;

exports[`makeTables should accept "all" inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "default",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeTables should accept complete inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "default",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "giant",
    "bottomRight": "giant",
    "topLeft": "giant",
    "topRight": "giant",
  },
}
`;

exports[`makeTables should accept dimensional inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "dotted",
      "width": "giant",
    },
    "left": Object {
      "style": "dotted",
      "width": "giant",
    },
    "right": Object {
      "style": "dotted",
      "width": "giant",
    },
    "top": Object {
      "style": "dotted",
      "width": "giant",
    },
  },
  "color": "primary",
  "cursor": "default",
  "font": "paragraph",
  "padding": Object {
    "bottom": "giant",
    "left": "giant",
    "right": "giant",
    "top": "giant",
  },
  "radius": Object {
    "bottomLeft": "xLarge",
    "bottomRight": "xLarge",
    "topLeft": "large",
    "topRight": "large",
  },
}
`;
